CustomCommand/ImportData/Console/Command/FileDataImporter.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace CustomCommand\ImportData\Console\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputOption;
+use CustomCommand\ImportData\Model\CustomerDataFactory;
+
+/**
+ * Class FileDataImporter
+ * @package CustomCommand\ImportData\Console\Command
+ */
+class FileDataImporter extends Command
+{
+    private const PROFILE = 'profile';
+
+    /**
+     * @var CustomerDataFactory
+     */
+    protected $dataFactory;
+
+    /**
+     * FileDataImporter constructor.
+     * @param CustomerDataFactory $dataFactory
+     */
+    public function __construct(CustomerDataFactory $dataFactory)
+    {
+        $this->dataFactory = $dataFactory;
+        parent::__construct();
+    }
+
+    /**
+     * Configure the command
+     */
+    protected function configure()
+    {
+        $options = [
+            new InputOption(
+                self::PROFILE,
+                '-s',
+                InputOption::VALUE_REQUIRED,
+                'Say the file extention'
+            ),
+        ];
+        $this->setName('customer:importer')
+            ->setDescription('Import customer data')
+            ->setDefinition($options)
+            ->addArgument('sourcePath', InputArgument::REQUIRED, 'Give source path');
+        parent::configure();
+    }
+
+    /**
+     * Execute the command
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $source = (string)$input->getArgument('sourcePath');
+        $profile = $input->getOption('profile');
+        $output->writeln("Profile: $profile");
+        $output->writeln("Source: $source");
+        if ($sourceExtension = pathinfo($source, PATHINFO_EXTENSION)) {
+            switch ($sourceExtension) {
+                case 'json':
+                    $jsonData = $this->readJsonData($source, $output);
+                    $this->processJsonData($jsonData);
+                    return 0;
+                    break;
+                case 'csv':
+                    $this->readCsvData($source, $output);
+                    return 0;
+                    break;
```

Comment:
It looks like you are only reading the CSV data and no further processing. Please review & modify the logic

Reply:
i have done but in a before saving to database